specification/batch/data-plane/Microsoft.Batch/2017-06-01.5.1/BatchService.json

```diff
@@ -13174,14 +13186,16 @@
           "items": {
             "type": "string"
           },
-          "title": "The list of task IDs that this task depends on. All tasks in this list must complete successfully before the dependent task can be scheduled."
+          "title": "The list of task IDs that this task depends on. All tasks in this list must complete successfully before the dependent task can be scheduled.",
+          "description": "The taskIds collection is limited to 64000 characters total (i.e. the combined length of all task ids). If the taskIds collection exceeds the maximum length, the Add Task request fails with error code TaskDependencyListTooLong. In this case consider using task id ranges instead."
         },
         "taskIdRanges": {
           "type": "array",
           "items": {
             "$ref": "#/definitions/TaskIdRange"
           },
-          "title": "The list of task ID ranges that this task depends on. All tasks in all ranges must complete successfully before the dependent task can be scheduled."
+          "title": "The list of task ID ranges that this task depends on. All tasks in all ranges must complete successfully before the dependent task can be scheduled.",
+          "description": "Because it is not possible to usefully specify a range of strings, tasks specified in a range must have ids that can be parsed as integers. For example, a task can depend on task ids 9 to 12, but cannot depend on task ids \"mytask\" to \"yourtask\"."
```

Comment:
>\"mytask\" to \"yourtask\" [](start = 248, length = 26)

for the future, perhaps, but this is kind of inherent in the type spec.  I think we reworked this in the C# docs to illustrate how an integer range mapped to stringy IDs - perhaps we should review and decide which we prefer.  Not needed for this port anyway